.github/workflows/build-and-test.yml

```diff
@@ -50,70 +65,71 @@ jobs:
       - name: Run Affected build
         run: yarn nx affected:build --exclude=docs --parallel=false
 
-      - run: yarn nx-cloud stop-all-agents
+      - run: npx nx-cloud stop-all-agents
 
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v3
         with:
-            token: ${{ secrets.CODECOV_TOKEN }}
+          token: ${{ secrets.CODECOV_TOKEN }}
 
   build-docs:
     runs-on: ubuntu-latest
     # Grant GITHUB_TOKEN the permissions required to make a Pages deployment
     permissions:
-      pages: write      # to deploy to Pages
-      id-token: write   # to verify the deployment originates from an appropriate source
+      actions: read
+      contents: read
+
+    steps:
+      - name: Checkout all commits
+        uses: actions/checkout@v3
+      - name: Set up dependencies
+        uses: ./.github/actions/setup
+        with:
+          # Docusaurus needs Node.js 16
+          node-version: 16
+
+      - name: Build docs
+        run: yarn nx build docs
+
+      - name: '[Merge] Upload docs'
+        if: github.ref == 'refs/heads/main'
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.docs-name }}
+          path: ${{ env.docs-path }}
+
+  deploy-docs:
+    name: '[Merge] Deploy docs'
+    runs-on: ubuntu-latest
+    # Grant GITHUB_TOKEN the permissions required to make a Pages deployment
+    permissions:
+      pages: write # to deploy to Pages
+      id-token: write # to verify the deployment originates from an appropriate source
+    if: github.ref == 'refs/heads/main'
 
     # Deploy to the github-pages environment
     environment:
       name: github-pages
       url: www.rx-angular.io
       #url: ${{ steps.deployment.outputs.page_url }}
 
-    env:
-      NX_CLOUD_DISTRIBUTED_EXECUTION: false
-      NX_DISTRIBUTED_TASK_EXECUTION: false
```

Comment:
I think this is still needed as we stop all agents in the build step, not in the build-docs step.